crates/ruff_linter/src/rules/pyupgrade/rules/printf_string_formatting.rs

```diff
@@ -286,6 +291,55 @@ fn clean_params_dictionary(right: &Expr, locator: &Locator, stylist: &Stylist) -
     Some(contents)
 }
 
+/// (In the case where the format string has only a single %-placeholder) and the righthand operand
+/// to % is a variable, try to offer a fix.
+///
+/// If the variable is not a tuple, then offer `({})`.
+/// ```python
+/// x = set(["hi", "hello"])
+/// print("%s" % x)
+/// print("{}".format(x))
+/// ```
+///
+/// If the variable is a 1-tuple, then offer `({}[0])`.
+/// ```python
+/// x = (1,)
+/// print("%s" % x)
+/// print("{}".format(x[0]))
+/// ```
+///
+/// If the variable is an n-tuple, the user has a bug in their code. Offer no fix.
+fn clean_params_variable<'a>(checker: &mut Checker, right: &Expr) -> Option<Cow<'a, str>> {
+    let Expr::Name(ast::ExprName { id, .. }) = right else {
+        return None;
+    };
+    let semantic = checker.semantic();
+    let binding_id = semantic.lookup_symbol(id.as_str())?;
+    let binding = semantic.binding(binding_id);
+    let rt = checker.locator().slice(right);
+    if let Some(expr) = analyze::typing::find_binding_value(binding, semantic) {
+        match expr {
+            // n-tuple
+            Expr::Tuple(ast::ExprTuple { elts, .. }) if 1 < elts.len() => {
```

Comment:
Nit: By convention I'd prefer the constant on the right-hand side of the operator. At first I misread this.

Reply:
Code won't exist in the next version that I push.